### PR TITLE
fix: handle empty Strava clubs payload

### PR DESCRIPTION
**Pull request by timothydowney**
## Summary
- treat an empty `clubs` array from `athlete.get()` the same as a missing clubs payload
- fall back to `athlete.listClubs()` before concluding WMV membership is absent
- add a regression test covering the empty-array response shape from Strava
- bump patch version to `0.14.8` and record the fix in the changelog

## Validation
- `cd server && npx jest src/__tests__/stravaClient.test.ts --runInBand --coverage=false`
- pre-commit checks: frontend and backend typecheck passed
- pre-commit checks: frontend and backend lint passed

## Notes
- this closes the last remaining behavior gap from the older draft PR #12 without reintroducing its superseded branch history
- Jest still prints the existing duplicate manual mock warning for `@google/genai`, but it did not block the targeted regression suite

### CODE REVIEW

File: server/src/stravaClient.ts [hunk 1]
```diff
@@ -205,8 +205,11 @@ async function getLoggedInAthlete(accessToken: string): Promise<any> {
     // Use the public API method provided by strava-v3 v4+.
     const athlete = await client.athlete.get({});
 
-    // If clubs aren't included on /athlete, fetch them explicitly and merge.
-    if (athlete && !Array.isArray((athlete as any).clubs)) {
+    // If clubs are omitted or empty on /athlete, fetch them explicitly and merge.
+    if (
+      athlete &&
+      (!Array.isArray((athlete as any).clubs) || (athlete as any).clubs.length === 0)
+    ) {
```
[the remark this answers is not in the document]
Comment (timothydowney) [reply server/src/stravaClient.ts]: Fixed in 17258cb. I pulled the `clubs` property into a local `athleteClubs` value, narrowed it with `Array.isArray`, and removed the repeated `any` casts from the fallback check.
